src/libsync/discovery.cpp

```diff
@@ -937,9 +937,14 @@ void ProcessDirectoryJob::processFileAnalyzeLocalInfo(
 
     // If it's not a move it's just a local-NEW
     if (!moveCheck()) {
-       postProcessLocalNew();
-       finalize();
-       return;
+        const bool isRootEncryptedFolderRename = base._isE2eEncrypted && base._e2eMangledName.isEmpty() && !base._path.contains('/');
```

Comment:
Are you sure about that `!base._path.contains('/')` part? I'd expect the problem to be for any encrypted folder which doesn't have an encrypted folder as parent, and those are not necessarily at the root of the exposed DAV file system. For instance you could have:
```
/
|- A <- NOT encrypted
|- B <- NOT encrypted
    |- C <- this one is encrypted
```

That's what concerned me in previous reviews with the term "root" being used, I feared such confusions. It's not really about it being at the top of the hierarchy or not, it's about it being the first one in the branch to be encrypted or not.

Reply:
@er-vin I haven't actually been thinking about such a scenario. Will have to re-test everything with root encrypted folders inside non-encrypted folders.